src/lib.rs

```diff
@@ -9,30 +9,33 @@
 // except according to those terms.
 
 #![no_std]
-#![cfg_attr(feature = "alloc", feature(alloc))]
+
+#![cfg_attr(feature = "nightly", feature(cfg_target_has_atomic))]
 
 #[cfg(test)]
 #[macro_use]
 extern crate std;
 
-#[cfg(feature = "alloc")]
```

Comment:
I missed this in the initial review but could you revert these changes? There are good reasons for wanting to avoid a dependency on the `alloc` crate, such as systems that don't have a global allocator. It's fine if the `alloc` feature is enabled by default, as long as there is a way to disable it.